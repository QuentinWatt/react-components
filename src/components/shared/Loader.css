.loader {
  width: 25px;
  aspect-ratio: 1;
  border-radius: 50%;
  border: 5px solid lightblue;
  border-right-color: black;
  animation: l2 1s infinite linear;
}

@keyframes l2 {
  to {
    transform: rotate(1turn);
  }
}
