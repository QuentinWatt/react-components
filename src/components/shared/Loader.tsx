import "./Loader.css";

interface Props {
  width?: string;
  ringWidth?: string;
  ringColor?: string;
  hilightColor?: string;
}

const Loader: React.FC<Props> = ({
  width = "28",
  ringWidth = "5",
  ringColor = "lightblue",
  hilightColor = "black",
}) => {
  const styles = {
    width: `${width}px`,
    borderWidth: `${ringWidth}px`,
    borderColor: `${ringColor}`,
    borderRightColor: `${hilightColor}`,
  };

  return <div className="loader" style={styles}></div>;
};

export default Loader;
