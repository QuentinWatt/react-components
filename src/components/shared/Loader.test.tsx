import { render, screen } from "@testing-library/react";
import "@testing-library/jest-dom";
import Loader from "./Loader";

jest.mock("./Loader.css", () => () => {});

describe("Shared loader component", () => {
  it("matches the snapshot", () => {
    const { asFragment } = render(<Loader />);
    expect(asFragment()).toMatchSnapshot();
  });

  it("has a loader div", () => {
    render(<Loader />);
    const element = screen.queryByTestId("loader");
    expect(element).toBeInTheDocument();
  });
});
