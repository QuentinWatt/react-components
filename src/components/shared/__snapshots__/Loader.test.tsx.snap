// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the snapshot 1`] = `
<DocumentFragment>
  <div
    class="loader"
    style="width: 28px; border-width: 5px; border-color: lightblue; border-right-color: black;"
  />
</DocumentFragment>
`;
